CoreScriptsRoot/CoreScripts/Topbar.lua

```diff
@@ -957,17 +979,42 @@ local function Create3DSettingsIcon(topBarInstance, panel, menubar)
 
 		BackgroundTransparency = 1,
 
-		Image = "rbxasset://textures/ui/Menu/hamburger3D.png"
+		Image = "rbxasset://textures/ui/VR/hamburger.png"
 	}
 
+	local menuState = false
+	GuiService.MenuOpened:connect(function() menuState = true end)
+	GuiService.MenuClosed:connect(function() menuState = false end)
+
+	function menuItem:OnSetActive(active)
+		MenuModule:SetVisibility(active)
+		if not active then
```

Comment:
I am not sure what is going on here:
* How come we are waiting for menuState to go false after setting the menu item to not active? Is there a spawn or a wait in the `MenuModule:SetVisibility` and is the function calling `OnSetActive` need to be assured that the menu is closed?
* This wait loop is concerning, we could break the corescripts if it sits around waiting forever
* We usually liked to put Async at the end of functions like this to let the caller know that it could stall

Ultimately, can we remove this wait and rather use an event to let the called know when it is closed?